molecule/commands/base.py

```diff
@@ -32,58 +32,28 @@ class InvalidHost(Exception):
 class BaseCommand:
     __metaclass__ = abc.ABCMeta
 
-    def __init__(self, command_args, global_args, molecule=False):
+    def __init__(self, command_args, args, molecule=False):
         """
         Initialize commands
 
         :param command_args: arguments of the command
-        :param global_args: arguments from the CLI
+        :param args: arguments from the CLI
         :param molecule: molecule instance
         """
         self.args = docopt.docopt(self.__doc__, argv=command_args)
         self.args['<command>'] = self.__class__.__name__.lower()
         self.command_args = command_args
 
-        self.static = False
+        self.molecule = core.Molecule(self.args)
+        self.main()
 
-        # give us the option to reuse an existing molecule instance
-        if not molecule:
-            self.molecule = core.Molecule(self.args)
-            self.molecule.main()
-        else:
-            self.molecule = molecule
```

Comment:
Yeah, I think that's why we were allowing the Converge command to pass in an instance of molecule instead of creating a new one for each command. You can see calls to converge still try to pass in self.molecule even though it's no longer being reused. I'm gonna do a following PR to re-enable this until we can examine it more closely.